### ISSUE TITLE
get_script_depends based on option

### ISSUE THREAD

**Issue by Stas0238** [elementor/elementor#16881]
Hello there !
I am writing here because it is almost 2 weeks after I wrote topic on Wordpress.org and no one answered me.

I would like to know if it is possible to make script depends based on option value ?
For example I have select control with options

```
$this->add_control(
  'event',
   [
	'label' => __( 'Event', 'st-domain' ),
	'type' => Controls_Manager::SELECT,
	'options' => [
	        'hover' => __( 'Hover', 'st-domain' ),
		'click' => __( 'Click', 'st-domain' ),
		'none' => __( 'None', 'st-domain' ),
	],
	'default' => 'hover',
	'frontend_available' => true,
    ]
);
```
and I have

```
public function get_script_depends() {
		$settings = $this->get_settings_for_display(); // or if I write $this->get_settings(); I will have the same results
		if ( $settings['event'] == 'click' ) {
			return ['custom_script'];
        }
		return [];
	}
```
but If I write it as above then after refresh Elementor editor cannot be loaded -> http://prntscr.com/1xmcyf1

My question is how is it possible to make depends only if one of options is choosen with specific value (in my case ‘event’ == ‘click’). I hope you understood me right.

Original Topic path -> https://wordpress.org/support/topic/get_script_depends-based-on-option/
Thanks

**Comment by Stas0238** [comment]
Thanks for 'fast' feedback.
As I see you are ignoring me everywhere 
https://wordpress.org/support/topic/get_style_depends-on-edit-mode-only/#post-15596311
https://wordpress.org/support/topic/get_script_depends-based-on-option/#post-15596300

Decided to provide you a video below of what I think checking if there are any possible good answer on my questions.
https://www.youtube.com/watch?v=vbb9MPYT23Y

**Comment by rami-elementor** [comment]
Hi @Stas0238 ,

If I understand correctly, you want to load the dependent script only if the user select some option in the control.

The short answer is - you can't do that.

But, you can change your JS to run some functionality only if the control has the value you want. I see you already expose the control value to JS by setting `'frontend_available' => true,` on your control.

**Comment by rami-elementor** [comment]
Check the documentation on [Frontend Available](https://developers.elementor.com/docs/controls/frontend-available/) values. You will find a good example there. It may help.

**Comment by Stas0238** [comment]
Hi @rami-elementor 

Yes I would like to load script file only in case some option is with specific value. (but for edit mode I could add kinda next workaround for loading script all time (in edit mode only) ) -> 
`public function get_script_depends() {

	    if ( \Elementor\Plugin::$instance->editor->is_edit_mode() || \Elementor\Plugin::$instance->preview->is_preview_mode() )
		    return [ 'my_script_key' ];

	    return [];

	}`

Your idea about checking the value in script if not perfrect because If I need to use this script only in case some specific value chosen then if it is not that value then file just loading without any sense.
Hope you understand my idea.

Also I found a workaround by adding `wp_enqueue_script('my_script_key')` in the `public function render()` just like next 
```
public function render() {

   $settings = $this->get_settings_for_display();

   if ( $settings['event'] == 'click' ) {

      wp_enqueue_script('my_script_key');

   }

   // other code 

}
```
Maybe my solution will make someone's day about searching it.

Also I heard you that it is not possible but anyway I decided to put my idea in your head :)

Regards, @Stas0238
